services/slam/builtin/builtin.go

```diff
@@ -369,7 +371,32 @@ func (slamSvc *builtIn) Position(ctx context.Context, name string) (*referencefr
 		return nil, errors.Wrap(err, "error getting SLAM position")
 	}
 
-	return referenceframe.ProtobufToPoseInFrame(resp.Pose), nil
+	pInFrame := referenceframe.ProtobufToPoseInFrame(resp.Pose)
+
+	// TODO DATA-531: Remove extraction and conversion of quaternion from the extra field in the response once the Rust SDK
+	// is available and the desired math can be implemented on the orbSLAM side
+	extra := resp.Extra.AsMap()
+
+	if val, ok := extra["quat"]; ok {
+		q := val.(map[string]interface{})
+
+		valTheta, ok1 := q["otheta"]
+		valX, ok2 := q["ox"]
+		valY, ok3 := q["oy"]
+		valZ, ok4 := q["oz"]
+
+		if !ok1 || !ok2 || !ok3 || !ok4 {
+			slamSvc.logger.Warnf("invalid format for quaternion returned, %v, skipping quaternion transform", q)
+			return pInFrame, nil
+		}
+		slamSvc.logger.Infof("valid format for quaternion returned, %v, performing quaternion transform", q)
+		ori := spatialmath.QuatToOV(quat.Number{Real: valTheta.(float64), Imag: valX.(float64), Jmag: valY.(float64), Kmag: valZ.(float64)})
+		newPose := spatialmath.NewPoseFromOrientation(pInFrame.Pose().Point(), ori)
+		pInFrame = referenceframe.NewPoseInFrame(pInFrame.FrameName(), newPose)
+
+	}
```

Comment:
[major] Agreed with John, might make sense to ask e.g. Peter L. for a quick glance at this.